internal/stream_push_helper.go

```diff
@@ -24,6 +27,52 @@ func (uploader *Uploader) PushStream(stream io.Reader) (string, error) {
 	return backupName, err
 }
 
+// TODO : unit tests
+// returns backup_prefix
+// (Note: individual parition names are built by adding '_0000.br' suffix)
+func (uploader *SplitStreamUploader) PushStream(stream io.Reader) (string, error) {
+	backupName := StreamPrefix + utility.TimeNowCrossPlatformUTC().Format(utility.BackupTimeFormat)
+
+	// Upload Stream:
+	var readers = splitmerge.SplitReader(stream, uploader.partitions, uploader.blockSize)
+
+	errCh := make(chan error)
+	defer close(errCh)
+	for i := 0; i < uploader.partitions; i++ {
+		dstPath := GetPartitionedStreamName(backupName, uploader.Compressor.FileExtension(), i)
+		tracelog.InfoLogger.Printf("Uploading... %v", dstPath)
+		go func(reader io.Reader) {
+			errCh <- uploader.PushStreamToDestination(reader, dstPath)
+		}(readers[i])
+	}
+
+	// Wait for upload finished:
+	var lastErr error
+	for i := 0; i < uploader.partitions; i++ {
+		err := <-errCh
+		if err != nil {
+			tracelog.WarningLogger.Printf("Failed to upload part of backup: %v", err)
+			lastErr = err
```

Comment:
Why not to abort immediately on the first error? You can utilise the `sync.ErrGroup` to cancel all goroutines on receiving error in one of them.